acme/examples/http01_example.py

```diff
@@ -0,0 +1,271 @@
+                                      PORT):
+            raise Exception('Verification failed')
+
+        # Let the CA server know that we are ready for the challenge.
+        client_acme.answer_challenge(challb, response)
+
+        # Wait for challenge status and then issue a certificate.
+        # It is possible to set a deadline time.
+        finalized_orderr = client_acme.poll_and_finalize(orderr)
+
+    return finalized_orderr.fullchain_pem
+
+
+# Main examples:
+
+
+def example_http():
+    """This example executes the whole process of fulfilling a HTTP-01
+    challenge for one specific domain.
+
+    The workflow consists of:
+    - Create account key.
+    - Register account and accept TOS.
+    - Select HTTP-01 within offered challenges by the CA server.
+    - Set up http challenge resource.
+    - Set up standalone web server.
+    - Create domain private key and CSR.
+    - Issue certificate.
+    - Change contact information.
+    - Renew Certificate
+    - Revoke Certificate
+    - Deactivate Account
+
+    """
+    logging.info('Example Challenge HTTP01')
+
+    client_acme = generate_client_account_key()
+
+    logging.info('Terms of Service URL: %s',
+                 client_acme.directory.meta.terms_of_service)
+
+    # Registration Resource: regr
+    # Creates account with contact information.
+    email = ('fake@example.com')
+    regr = client_acme.new_account(
+        messages.NewRegistration.from_data(
+            email=email, terms_of_service_agreed=True))
+
+    logging.info('Account registered.')
+
+    pkey_pem, csr_pem = new_csr_comp(DOMAIN)
+    orderr = client_acme.new_order(csr_pem)
+
+    challb = select_http01_chall(orderr)
+
+    # The certificate is ready to be used in the variable "fullchain_pem".
+    fullchain_pem = perform_http01(client_acme, challb, orderr)
+
+    logging.info('Certificate issued: \n%s', fullchain_pem)
+
+    regr = verify_registration(client_acme, regr)
+
+    # Change contact information
+    email = 'newfake@example.com'
+    regr = client_acme.update_registration(
+        regr.update(
+            body=regr.body.update(
+                contact=('mailto:' + email,)
+            )
+        )
+    )
+    logging.info('New contact info: %s', repr(regr.body.contact))
+
+    logging.info('Renew Certificate')
+
+    _, csr_pem = new_csr_comp(DOMAIN, pkey_pem)
+
+    orderr = client_acme.new_order(csr_pem)
+
+    challb = select_http01_chall(orderr)
+
+    logging.info('Performing challenge')
+
+    fullchain_pem = perform_http01(client_acme, challb, orderr)
+
+    logging.info('Certificate renewed: \n%s', fullchain_pem)
+
+    logging.info('Revoke and Deactivate')
+
+    fullchain_com = jose.ComparableX509(
+        OpenSSL.crypto.load_certificate(
+            OpenSSL.crypto.FILETYPE_PEM, fullchain_pem))
+
+    try:
+        client_acme.revoke(fullchain_com, 0)  # revocation reason = 0
+    except errors.ConflictError:
```

Comment:
Very helpful to show the errors you get, love that you do this here and in a couple other places!